### PR TITLE
New product database

### PR DESCRIPTION
**Pull request by isavelasq**
Hello please review the way I set up adding the data to the database, and if there are any bugs or formatting issues.

### CODE REVIEW

**Comment by aarondme** [comment]
That'll work. I'd move it to the database folder. Also if you'd like you could extend the Database class and use the put() method instead to make the code a bit more uniform. (See AccountDatabase for an example)

File: app/src/main/java/com/example/b07group7project/database/SaveProductImplementation.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+public class SaveProductImplementation extends Database implements SaveProduct {
+
+    private DatabaseReference productsRef;
+
+    @Override
+    public void saveProductToFirebase(String itemName, String itemDesc, String itemURL, String storeID, double itemPrice) {
+        String itemID = UUID.randomUUID().toString(); // Generate a unique key for the product
+
+        // Save the product information to the Products node
+        productsRef = root.child("Products")
```
Comment (aarondme) [inline app/src/main/java/com/example/b07group7project/database/SaveProductImplementation.java:16]: change to Constants.products

File: app/src/main/java/com/example/b07group7project/database/SaveProductImplementation.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                .child(storeID)
+                .child(itemID);
+
+        // Create a HashMap to store the product information
+        Map<String, Object> productInfo = new HashMap<>();
+        productInfo.put("name", itemName);
```
Comment (aarondme) [inline app/src/main/java/com/example/b07group7project/database/SaveProductImplementation.java:22]: use the values from Constants class as keys

**Review by aarondme — CHANGES_REQUESTED** [review]

**Review by aarondme — APPROVED** [review]
Lgtm, tested it and added an item successfully. A toast when the fields are cleared on screen would be nice

**Review by HariBala246 — APPROVED** [review]
Cool

**Pull request merged**